pipeline/stage03_trigger_detection/scripts/minima.py

```diff
@@ -1,29 +1,53 @@
 import neo
 import numpy as np
 import quantities as pq
-from scipy.signal import argrelmin
+from scipy.signal import argrelmin, argrelmax, find_peaks
 import argparse
 from distutils.util import strtobool
 from utils import load_neo, write_neo, remove_annotations
 
 
-def detect_minima(asig, order, interpolation_points, interpolation):
+def detect_minima(asig, order, interpolation_points, interpolation, threshold_fraction,  Min_Peak_Distance):
+        
     signal = asig.as_array()
+    times = asig.times
     sampling_time = asig.times[1] - asig.times[0]
-
-    t_idx, channel_idx = argrelmin(signal, order=order, axis=0)
-
+    min_idx, channel_idx_minima = argrelmin(signal, order=order, axis=0)
```

Comment:
using all minima with `argrelmin` for the algorithm results in an unfavorable minima selection:
![find_minima_example](https://user-images.githubusercontent.com/16289604/125846994-6f247341-f863-4d95-8b42-f92ade1daadf.png)

using instead `find_peaks(-channel_signal, height=threshold[channel])` yields better results. 
However, then there is no mechanism for selection all minima... using separate thresholds for minima and peaks would work, but maybe there is a more elegant solution

Reply:
We should definitely think about it.